app/src/main/java/ar/scacchipa/twittercloneapp/domain/usecase/AuthorizationUseCase.kt

```diff
@@ -1,21 +1,29 @@
 package ar.scacchipa.twittercloneapp.domain.usecase
 
-import ar.scacchipa.twittercloneapp.data.repository.IAuthorizationRepository
+import ar.scacchipa.twittercloneapp.data.repository.ICredentialRepository
+import ar.scacchipa.twittercloneapp.domain.model.Credential
 import ar.scacchipa.twittercloneapp.domain.model.ResponseDomain
 import ar.scacchipa.twittercloneapp.utils.Constants
 
 open class AuthorizationUseCase(
-    private val repository: IAuthorizationRepository
+    private val credentialRepository: ICredentialRepository
 ) {
-    open suspend operator fun invoke(
+    suspend operator fun invoke(
         transitoryToken: String
     ): ResponseDomain {
-        return repository.requestAccessToken(
+        val accessTokenRequest = credentialRepository.getExternalCredential(
             transitoryToken = transitoryToken,
             grantType = Constants.GRANT_TYPE_AUTHORIZATION_CODE,
             clientId = Constants.CLIENT_ID,
             redirectUri = Constants.REDIRECT_URI,
             codeVerifier = Constants.CODE_VERIFIER_CHALLENGE,
-            state = Constants.STATE_STATE)
+            state = Constants.STATE_STATE
```

Comment:
maybe just leave it as Constants.STATE it's enough, no need to be redundant since there is no other variable called STATE or related